src/main/java/com/example/todoz/controllers/AuthenticationController.java

```diff
@@ -47,14 +47,26 @@ public String register(RegisterDTO registerDTO, RedirectAttributes ra) {
             ra.addFlashAttribute("userExists", true);
             return "redirect:/register";
         }
-        else {
+
+        if (validateUsername(registerDTO.username()) && validatePassword(registerDTO.password())){
             User user = new User();
             user.setUsername(registerDTO.username());
             user.setPassword(passwordEncoder.encode(registerDTO.password()));
             user.setPussyMeter(registerDTO.pussyMeter());
             userService.save(user);
+        } else {
+            ra.addFlashAttribute("usernameWrong", true);
+            return "redirect:/register";
         }
 
         return "redirect:/login";
     }
+
+    public boolean validateUsername(String username) {
+        return !username.isBlank() && username.length() >= 3;

```

Comment:
password should not generally contain " " and this method doesn't check it

Reply:
whitespace validation added)